examples/controls_example/public/react_controls/data_controls/data_control_editor.test.tsx

```diff
@@ -0,0 +1,236 @@
+import { TimeRange } from '@kbn/es-query';
+import { I18nProvider } from '@kbn/i18n-react';
+import { act, fireEvent, render, RenderResult, waitFor } from '@testing-library/react';
+
+import { registerControlFactory } from '../control_factory_registry';
+import { ControlGroupApi } from '../control_group/types';
+import { DataControlEditor } from './data_control_editor';
+import {
+  getMockedOptionsListControlFactory,
+  getMockedRangeSliderControlFactory,
+  getMockedSearchControlFactory,
+} from './mocks/data_control_mocks';
+import { DEFAULT_CONTROL_GROW, DEFAULT_CONTROL_WIDTH } from '@kbn/controls-plugin/common';
+import { DataControlEditorState } from './open_data_control_editor';
+
+const mockDataViews = dataViewPluginMocks.createStartContract();
+const mockDataView = createStubDataView({
+  spec: {
+    id: 'logstash-*',
+    fields: {
+      ...stubFieldSpecMap,
+      'machine.os.raw': {
+        name: 'machine.os.raw',
+        customLabel: 'OS',
+        type: 'string',
+        esTypes: ['keyword'],
+        aggregatable: true,
+        searchable: true,
+      },
+    },
+    title: 'logstash-*',
+    timeFieldName: '@timestamp',
+  },
+});
+mockDataViews.get = jest.fn().mockResolvedValue(mockDataView);
+
+const dashboardApi = {
+  timeRange$: new BehaviorSubject<TimeRange | undefined>(undefined),
+  lastUsedDataViewId$: new BehaviorSubject<string>(mockDataView.id!),
+};
+const controlGroupApi = {
+  parentApi: dashboardApi,
+  grow: new BehaviorSubject(DEFAULT_CONTROL_GROW),
+  width: new BehaviorSubject(DEFAULT_CONTROL_WIDTH),
+} as unknown as ControlGroupApi;
+
+describe('Data control editor', () => {
+  const mountComponent = async ({
+    initialState,
+  }: {
+    initialState?: Partial<DataControlEditorState>;
+  }) => {
+    mockDataViews.get = jest.fn().mockResolvedValue(mockDataView);
+
+    const controlEditor = render(
+      <I18nProvider>
+        <DataControlEditor
+          onCancel={() => {}}
+          onSave={() => {}}
+          parentApi={controlGroupApi}
+          initialState={{
+            dataViewId: dashboardApi.lastUsedDataViewId$.getValue(),
+            ...initialState,
+          }}
+          services={{ dataViews: mockDataViews }}
+        />
+      </I18nProvider>
+    );
+
+    await waitFor(() => {
+      expect(mockDataViews.get).toHaveBeenCalledTimes(1);
+    });
+
+    return controlEditor;
+  };
+
+  const selectField = async (controlEditor: RenderResult, fieldName: string) => {
+    expect(controlEditor.queryByTestId(`field-picker-select-${fieldName}`)).toBeInTheDocument();
+    await act(async () => {
+      fireEvent.click(controlEditor.getByTestId(`field-picker-select-${fieldName}`));
+    });
+  };
+
+  const getPressedAttribute = (controlEditor: RenderResult, testId: string) => {
+    return controlEditor.getByTestId(testId).getAttribute('aria-pressed');
+  };
+
+  beforeAll(() => {
+    /** Register all of my mocked data controls */
+    registerControlFactory('search', async () => {
```

Comment:
Does `registerControlFactory` pollute control registry for other tests? Would it be better to just mock `control_factory_registry` methods directly?

Reply:
Good call - done in 1ef8e39f72192f9ee754364b0d0d59b31e435995